src/NuGet.Clients/NuGet.Console/PowerConsoleCommands.cs

```diff
@@ -0,0 +1,187 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Shell;
+using NuGetConsole.Implementation;
+using System.ComponentModel.Design;
+using NuGetConsole.Implementation.PowerConsole;
+using Microsoft.VisualStudio.Threading;
+using Microsoft.VisualStudio.ComponentModelHost;
+using NuGet.VisualStudio;
+using System.Runtime.InteropServices;
+
+namespace NuGetConsole
+{
+    public class PowerConsoleCommands
+    {
+        public static PowerConsoleCommands Instance
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Get VS IComponentModel service.
+        /// </summary>
+        private IComponentModel ComponentModel
+        {
+            get; 
+        }
+
+        private PowerConsoleWindow PowerConsoleWindow
+        {
+            get;
+        }
+
+        private HostInfo ActiveHostInfo => PowerConsoleWindow.ActiveHostInfo;
+
+        private IWpfConsole WpfConsole
+        {
+            get;
+        }
+
+        public static async System.Threading.Tasks.Task InitializeAsync(AsyncPackage package)
+        {
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync(package.DisposalToken);
+            var cmdSvc = await package.GetServiceAsync(typeof(IMenuCommandService)) as OleMenuCommandService;
+            var comSvc = await package.GetComponentModelAsync();
+
+            Instance = new PowerConsoleCommands(cmdSvc, comSvc);
+        }
+
+        private PowerConsoleCommands(OleMenuCommandService commandService, IComponentModel comModelSvc)
+        {
+            if (commandService == null)
+            {
+                throw new ArgumentNullException(nameof(commandService));
+            }
+
+            // init services
+            ComponentModel = comModelSvc;
+            PowerConsoleWindow = ComponentModel.GetService<IPowerConsoleWindow>() as PowerConsoleWindow;
```

Comment:
Could you have moved all this from constructor to `InitializeAsync`, and use `GetServiceAsync` here as well?

Reply:
I chatted with him. 
It seems he is using [this pattern](https://stackoverflow.com/a/12520574) for creating Singleton with async method.

Also IComponentModel doesn't have GetServiceAsync method.
![image](https://user-images.githubusercontent.com/8766776/99432824-8def4700-28c1-11eb-83d6-ad50465f4e4e.png)

Reply:
@srdjanjovcic Can we use `AsyncServiceProvider.GlobalProvider.GetServiceAsync()` to get MEF objects, as we do with `componentModel.GetService()` ?

Reply:
It doesn't work with AsyncServiceProvider.GlobalProvider.GetServiceAsync(), I already tested it.
![image](https://user-images.githubusercontent.com/8766776/99557266-580c9a00-2977-11eb-9866-a1d9740b3f64.png)

When can we use AsyncServiceProvider.GlobalProvider.GetServiceAsync() and when not that is the question?

Reply:
> @srdjanjovcic Can we use `AsyncServiceProvider.GlobalProvider.GetServiceAsync()` to get MEF objects, as we do with `componentModel.GetService()` ?

@dominoFire @srdjanjovcic 

It doesn't work with AsyncServiceProvider.GlobalProvider.GetServiceAsync(), I already tested it.
![image](https://user-images.githubusercontent.com/8766776/99557266-580c9a00-2977-11eb-9866-a1d9740b3f64.png)

When can we use AsyncServiceProvider.GlobalProvider.GetServiceAsync() and when not that is the question?

Reply:
MEF (`ComponentModel`) was something that was a generic .NET library, but there's also a VS specific implementation, I guess to improve some limitations of the generic MEF implementations.

Visual Studio's service provider is something else: https://docs.microsoft.com/en-us/visualstudio/extensibility/using-and-providing-services?view=vs-2019

The way the two register services are different, and you have to use the one that contains the service you're looking for. If the object you want was registered in MEF, you must use MEF to get it. If the object you want was registered through VS's service definition, you must use VS's service provider.

They work somewhat differently (have quite different features), so I guess that's why objects defined in one is not available in the other.

Reply:
Removed whole this newly added file, because it was not beneficial.